LibreNMS/Data/Store/Kafka.php

```diff
@@ -0,0 +1,326 @@
+        if ($this->kafkaFlushTimeout == null) {
+            $this->kafkaFlushTimeout = Config::get('kafka.flush.timeout', 100);
+        }
+    }
+
+    public function __destruct()
+    {
+        $this->terminate();
+    }
+
+    public function terminate(): void
+    {
+        // Safely flush the producer to ensure all messages are sent before shutdown
+        $this->safeFlush();
+    }
+
+    public static function getClient(): Producer
+    {
+        $conf = new Conf();
+        // Set the log level
+        $conf->set('log_level', (string) LOG_DEBUG);
+        // Set the log callback for exceptions
+        $conf->setDrMsgCb(
+            function (Producer $producer, Message $message): void {
+                if ($message->err !== RD_KAFKA_RESP_ERR_NO_ERROR) {
+                    error_log($message->errstr());
+                }
+            }
+        );
+        // Set the log callback for logs
+        $conf->setLogCb(
+            function (Producer $producer, int $level, string $facility, string $message): void {
+                error_log('KAFKA: ' . $message);
+            }
+        );
+
+        // Set the kafka broker servers
+        $conf->set('bootstrap.servers', Config::get('kafka.broker.list', '127.0.2.2:9092'));
+        // Set the idempotence
+        $conf->set('enable.idempotence', Config::get('kafka.idempotence', false) ? 'true' : 'false');
+        // Max queue allowed messages in poller memory
+        $conf->set('queue.buffering.max.messages', Config::get('kafka.buffer.max.message', 1_000));
+        // Num of messages each call to kafka
+        $conf->set('batch.num.messages', Config::get('kafka.batch.max.message', 200));
+        // Max wait time to acumulate before sending the batch
+        $conf->set('linger.ms', Config::get('kafka.linger.ms', default: 50));
+        // Change ACK
+        $conf->set(
+            'request.required.acks',
+            // If idempotence is enabled, set to 'all' to ensure all messages are acknowledged
+            // Otherwise, use the configured value or default to '1'
+            // '1' means the leader will acknowledge the message, 'all' means all replicas must acknowledge
+            Config::get('kafka.idempotence', false) ? 'all' :
+                (Config::get('kafka.request.required.acks', '-1'))
+        );
+
+        // check if debug for ssl was set and enable it
+        $confKafkaSSLDebug = Config::get('kafka.security.debug', null);
+        $confKafkaSSLDebug != null || strlen($confKafkaSSLDebug) !== 0 ? $conf->set('debug', $confKafkaSSLDebug) : null;
+
+        // config ssl
+        $isSslEnabled = Config::get('kafka.ssl.enable', false);
+        if ($isSslEnabled) {
+            $conf->set('security.protocol', Config::get('kafka.ssl.protocol', 'ssl'));
+            $conf->set('ssl.endpoint.identification.algorithm', 'none');
+
+            // prepare all necessary librenms kafka config with associated rdkafka key
+            $kafkaSSLConfigs = [
```

Comment:
I find some of these SSL settings odd.  I'm not sure why they are needed.  Is this running it's own server?